### PR TITLE
Use a local branch to create a scan. (AST-75824)

### PR DESCRIPTION
**Pull request by sarahCx**
By submitting a PR to this repository, you agree to the terms within the [Checkmarx Code of Conduct](../docs/code_of_conduct.md). Please see the [contributing guidelines](../docs/contributing.md) for how to create and submit a high-quality PR for this repo.

### Description

"Adding functionality to use a local Git branch when creating a scan, so that it's possible to create a scan on a new branch."

### References

https://checkmarx.atlassian.net/browse/AST-75824

### Testing

Adding an integration test to validate: 
1. select project and get local branch.
2. exist local branch in branches list.
3. not get a scans with local branch.
4. run scan with local branch.
5. select project and get branch main with scan

### Checklist

- [ ] I have added documentation for new/changed functionality in this PR (if applicable).
- [ ] All active GitHub checks for tests, formatting, and security are passing
- [ ] The correct base branch is being used

### CODE REVIEW

File: src/utils/pickers/pickers.ts [hunk 1]
```diff
@@ -31,15 +32,35 @@ export async function projectPicker(
...
 }
 
+async function setDefaultBranch(
+  projectId: string,
+  context: vscode.ExtensionContext,
+  logs: Logs,
+) {
+  try {
+    const gitApi = await getGitAPIRepository();
+    const gitBranchName = gitApi.repositories[0]?.state.HEAD?.name; // TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
```
Comment (Korjen97) [inline src/utils/pickers/pickers.ts:51]: Lines 50-51 appear more than once in the entire PR, mabye try to use a utility function like :
async function getCurrentGitBranch() {
const gitApi = await getGitAPIRepository();
return gitApi.repositories[0]?.state.HEAD?.name;
}

File: src/views/resultsView/createScanProvider.ts [hunk 2]
```diff
@@ -53,6 +53,14 @@ async function createScanForProject(
   logs: Logs
 ) {
   const scanBranch: Item = context.workspaceState.get(constants.branchIdKey);
+  if(scanBranch.id === constants.localBranch){
+    const gitApi = await getGitAPIRepository();
+    const gitBranchName = gitApi.repositories[0]?.state.HEAD?.name;//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
```
Comment (AlvoBen) [inline src/views/resultsView/createScanProvider.ts:58]: Also here:
//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly

File: src/utils/pickers/pickers.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (sarahCx) [reply src/utils/pickers/pickers.ts]: No, we need to wait for the branch change listener.

File: src/views/resultsView/createScanProvider.ts (hunk 2 above)
Comment (sarahCx) [reply src/views/resultsView/createScanProvider.ts]: No, we need to wait for the branch change listener.

**Review by AlvoBen — APPROVED** [review]

**Pull request merged**